### PR TITLE
Not require "Directory.Read.All" scope if "azure.activedirectory.user-group.allowed-group-names" not configured.

### LINKED ISSUE #21284: [FEATURE REQ] Not require "Directory.Read.All" scope if "azure.activedirectory.user-group.allowed-group-names" not configured.
If we do not need group names, we can remove "Directory.Read.All" scope.

Refs: https://github.com/Azure/azure-sdk-for-java/issues/21162#issuecomment-837035562

### PR DESCRIPTION
**Pull request by chenrujun**
Fix #21284

### CODE REVIEW

**Comment by chenrujun** [comment]
Hi, @moarychan , @zhichengliu12581 , @yiliuTo , please help to review this PR when you have time.

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/AADClientRegistrationRepository.java [hunk 1]
```diff
@@ -199,21 +199,31 @@ public static int resourceServerCount(Set<String> scopes) {
...
-        if (properties.allowedGroupIdsConfigured() || properties.allowedGroupNamesConfigured()) {
-            // The 2 scopes are need to get group name from graph.
-            result.add(properties.getGraphBaseUri() + "User.Read");
+        // About "Directory.Read.All" and "User.Read", please refer to:
+        // 1. https://docs.microsoft.com/en-us/graph/permissions-reference
+        // 2. https://github.com/Azure/azure-sdk-for-java/issues/21284#issuecomment-888725241
+        if (properties.allowedGroupNamesConfigured()) {
+            // "Directory.Read.All" allows to get group id and group name.
             result.add(properties.getGraphBaseUri() + "Directory.Read.All");
+        } else if (properties.allowedGroupIdsConfigured()) {
```
Comment (moarychan) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/AADClientRegistrationRepository.java:208]: Is it possible both scopes need?

Comment (chenrujun) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/AADClientRegistrationRepository.java:208]: No. `User.Read` is not necessary anymore we have `Directory.Read.All`.

Please refer to the link in line 204.

Comment (chenrujun) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/AADClientRegistrationRepository.java:208]: @moarychan , could you  please comment `LGTM` if you think this PR is OK to merge?

**Comment by moarychan** [comment]
LGTM.

**Comment by chenrujun** [comment]
Hi, @yiliuTo , could you please help to approve this PR?

**Review by yiliuTo — APPROVED** [review]
LGTM

**Pull request merged**